scripts/aws/monitor.py

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/python
+#
+# Run this script from the directory its in
+# This script requires following arguments
+# --aws-profile (name of the profile to use for aws credentials)
+# --api-gw (name of the api_gateway)
+# --splunk-host (splunk host IP and port)
+# --splunk-token (splunk access token)
+# e.g.
+# python monitor.py --aws-profile test --api-gw api --splunk-host 10.2.1.2:99 --splunk-token xxx
+#
+
+import logging
+import time
+import argparse
+import tempfile
+import shutil
+from zipfile import ZipFile
+from os.path import os, basename
+import botocore.session
+import botocore.exceptions
+from jinja2 import Environment, FileSystemLoader
+
+
+def create_api_alarm(cw_session, alarm_name, metric,
+                     namespace, stat, comparison, description,
+                     threshold, period, eval_period, dimensions, topic):
+    """Create the alarm for appropriate metric in API Gateway"""
+    response = cw_session.put_metric_alarm(
+        AlarmName=alarm_name,
+        AlarmDescription=description,
+        ActionsEnabled=True,
+        Namespace=namespace,
+        Dimensions=dimensions,
+        MetricName=metric,
+        Statistic=stat,
+        Period=period,
+        EvaluationPeriods=eval_period,
+        Threshold=threshold,
+        ComparisonOperator=comparison,
+        AlarmActions=[topic])
+
+    logging.info('response for creating api alarm = "%s"',
+                 response['ResponseMetadata'])
+
+
+def create_lambda_function_zip(temp_dir, splunk_host, splunk_token):
+    """Updates and Zips the lambda function file"""
+
+    j2_env = Environment(loader=FileSystemLoader(os.path.join(os.path.dirname(__file__), 'templates')),
+                         trim_blocks=False)
+    splunk_values = {
+        'splunk_ip': splunk_host,
+        'token': splunk_token,
+    }
+    js_file = temp_dir + '/index.js'
+    with open(js_file, 'w') as lambda_function_file:
+        lf_data = j2_env.get_template('index.j2').render(splunk_values)
```

Comment:
small think, but let's make the template index.js.j2, this way it's clear this is a jinja template of a js file.